pkg/install/stack/usercluster-mla/validation.go

```diff
@@ -20,16 +20,119 @@ import (
+	// if the user has configured this or any older version explicitly in their Helm values,
+	// we do not need to perform any further checks and can save a lot of work.
+	minioTag, ok := opt.HelmValues.GetString(yamled.Path{"minio", "image", "tag"})
+	if ok {
+		if !strings.HasPrefix(minioTag, "RELEASE.") {
+			opt.Logger.WithField("tag", minioTag).Warn("Cannot parse customized Minio tag, cannot skip PVC compatibility check")
+		} else if minioTag <= lastCompatibleMinioRelease {
+			return nil // an old release is configured, nothing can go wrong
+		}
+	}
+
+	release, err := opt.HelmClient.GetRelease(MinioNamespace, MinioReleaseName)
+	if err != nil {
+		return fmt.Errorf("failed to check Helm releases: %w", err)
+	}
+
+	// Minio has not been installed (yet?); perfect, the user is free to
+	// install whatever version they wish.
+	if release == nil {
+		return nil
+	}
+
+	// Checking compatibility requires to actually inspect Minio's filesystem;
+	// the current Helm release version won't tell us the original version that
+	// created the storage, and Minio's Admin API does not provide the filesystem
+	// driver name.
+	pods := corev1.PodList{}
+	if err := opt.KubeClient.List(ctx, &pods, &ctrlruntimeclient.ListOptions{
+		Namespace: MinioNamespace,
+		LabelSelector: labels.ValidatedSetSelector{
+			"app": "minio",
+		},
+	}); err != nil {
+		return fmt.Errorf("failed to find Minio pod: %w", err)
+	}
+
+	// As the Helm chart provisions a singular PVC, we expect a singular pod. There
+	// is no option in the chart to configure a GCS/S3-backed Minio proxy that might
+	// run multiple replicas.
+	if len(pods.Items) != 1 {
+		return fmt.Errorf("expected exactly 1 Minio Pod, but found %d; cannot exec and check PVC contents", len(pods.Items))
+	}
+
+	minioPod := ctrlruntimeclient.ObjectKeyFromObject(&pods.Items[0])
+
+	// Exec into the pod and look under Minio's hood.
+	command := []string{"cat", "/storage/.minio.sys/format.json"}
+	stdout, _, err := podexec.ExecuteCommand(ctx, opt.RestConfig, minioPod, "minio", command...)
+	if err != nil {
+		return fmt.Errorf("failed to execute command in Minio container: %w", err)
+	}
+
+	// parse Minio's config file
+	type minioFormat struct {
+		Format string `json:"format"`
+	}
+
+	data := minioFormat{}
+	if err := json.Unmarshal([]byte(stdout), &data); err != nil {
+		return fmt.Errorf("failed to decode %q as JSON: %w", stdout, err)
+	}
+
+	// Bad news: This Minio is using the old, legacy fs driver and needs to be migrated manually.
+	if data.Format == "fs" {
+		return errors.New("the Minio storage is using the `fs` filesystem driver, which is incompatible with more recent Minio releases and requires a migration; please refer to https://docs.kubermatic.com/kubermatic/v2.23/installation/upgrading/upgrade-from-2.22-to-2.23/#minio-upgrade for more information")
+	}
+
+	// Good news, the storage is probably using "xl" and so it's future-ready.
```

Comment:
tiny nit: I read this comment in a way that we expect it to be "xl", but it could be something differently. Therefore I think it makes sense to have something like this

```Go
if data.Format != "fs" && data.Format != "xl" {
	// return some error about unknown format
}
```